testing_exercise/testing.js

```diff
@@ -0,0 +1,38 @@
+function replaceWith(string, pattern, replacer) {
+  var newString = [];
```

Comment:
whenever you find yourself creating an array, then using a forEach to push to that array, consider whether you could use a map instead.